core/src/main/java/xtdb/vector/ValueVectorReader.java

```diff
@@ -77,9 +81,14 @@ public Field getField() {
         return vector.getField();
     }
 
+    protected byte[] getRawBytes(int idx) {
+        return HashUtilKt.byteBufferToBytes(getBytes(idx));
+    }
+
     @Override
-    public int hashCode(int idx, ArrowBufHasher hasher) {
-        return vector.hashCode(idx, hasher);
+    public int hashCode(int idx, XtByteHasher hasher) {
+        if (isNull(idx)) return ArrowBufPointer.NULL_HASH_CODE;
+        return hasher.hashValue(getRawBytes(idx));
     }
```

Comment:
My idea was that for all vector types for which we don't specialize `hashCode` we could default to hashing the raw bytes. This would mean that for example `:keyword` and `:string` still hash to the same value, but they get filtered afterwards by the EE. The other option is to use the vector type + raw data in the default hashing. The main reason I was thinking of `getRawBytes` was so that no objects get created.